### PR TITLE
Fix spans for E094, E104, E106, E111, and E119

### PR DESCRIPTION
**Pull request by ahmafi**
Fix for issue #501

### CODE REVIEW

**Comment by ahmafi** [comment]
I have read and hereby agree to quick-lint-js' Contributor License Agreement (CLA-v1.md).

**Comment by ahmafi** [comment]
rechek

File: src/quick-lint-js/parse.h [hunk 1]
```diff
@@ -2481,9 +2484,9 @@ class parser {
     bool parsed_body =
         this->parse_and_visit_statement_disallowing_declaration(v);
     if (!parsed_body) {
+      const char8 *here = this->lexer_.end_of_previous_token();
       this->error_reporter_->report(error_missing_body_for_for_statement{
```
Comment (strager) [inline src/quick-lint-js/parse.h:2488]: Perhaps we should include a *NOTE* mentioning where the `for` loop starts. See E141 for example.

**Review by strager — APPROVED** [review]
Looks great!

Can you write a summarizing commit message, or improve each of your commit's messages? "fix span E104 " doesn't explain *why* the change is being made.

**Comment by strager** [comment]
Merged as commits bdeadfef (nice) through 06bc76f6.

**Pull request closed without merging**